woocommerce/woocommerce-ios
# IPP Analytics: add `country` property to typed IPP events

Part of #5984 


### Description


As part of the tracking plan for Canada IPP support in pdfdoF-vx-p2, we're adding `country` property among others to all relevant IPP events including the pre-existing events not specific to Canada support. Since there are many affected events, I started with events defined in the `WooAnalyticsEvent.InPersonPayments` enum that are already typed. There are still other events like `*_card_present_onboarding_*` and `*_receipt_*` events that I will add the `country` property in a separate PR by also adding these events to the `WooAnalyticsEvent.InPersonPayments` enum.

There are 14 events updated in this PR:
- `card_reader_discovery_failed`
- `card_reader_connection_success`
- `card_reader_connection_failed`
- `card_reader_disconnect_tapped`
- `card_reader_software_update_tapped`
- `card_reader_software_update_started`
- `card_reader_software_update_failed`
- `card_reader_software_update_success`
- `card_reader_software_update_cancel_tapped`
- `card_reader_software_update_canceled`
- `card_present_collect_payment_tapped`
- `card_present_collect_payment_failed`
- `card_present_collect_payment_canceled`
- `card_present_collect_payment_success`

This PR also contains the removal of some unused code or unnecessary `self` usage and typo fixes from browsing the relevant code.

### Testing instructions


Feel free to test as many events as possible, I tested all events in two stores in the US and Canada in simulator with simulated card reader enabled and also on a device with an M2 reader.

For each event, there should be a new `country` property that shows the country code of the store.

- `card_reader_discovery_failed`: you can turn off Bluetooth on a device to trigger this event
- `card_reader_connection_success`
- `card_reader_connection_failed`
- `card_reader_disconnect_tapped`
- `card_reader_software_update_tapped`: this is easier to test in a simulator with simulated card reader enabled
- `card_reader_software_update_started`
- `card_reader_software_update_failed`: you can turn off Wifi in the middle of the software update to trigger this event
- `card_reader_software_update_success`
- `card_reader_software_update_cancel_tapped`
- `card_reader_software_update_canceled`
- `card_present_collect_payment_tapped`
- `card_present_collect_payment_failed`
- `card_present_collect_payment_canceled`
- `card_present_collect_payment_success`

---
